### PR TITLE
Add determinant localization approximation (DLA)

### PR DESCRIPTION
**Pull request by ye-luo**
From https://arxiv.org/abs/1907.04432. @zenandrea please have a look
still need
- [x] manual entry
- [x] unit test

@PDoakORNL Please have a look if I use enum in a good way or not.

### CODE REVIEW

**Comment by prckent** [comment]
@PDoakORNL Are you OK with this code?

File: src/QMCWaveFunctions/TrialWaveFunction.cpp [hunk 1]
```diff
@@ -335,6 +333,28 @@ TrialWaveFunction::ValueType TrialWaveFunction::full_ratio(ParticleSet& P, int i
...
 
+TrialWaveFunction::RealType TrialWaveFunction::calcRatio(ParticleSet& P, int iat, ComputeType ct)
+{
+  ValueType r(1.0);
+  std::vector<WaveFunctionComponent*>::iterator it(Z.begin());
+  std::vector<WaveFunctionComponent*>::iterator it_end(Z.end());
+  for (int ii = V_TIMER; it != it_end; ++it, ii += TIMER_SKIP)
+  {
+    myTimers[ii]->start();
+    if (ct == ComputeType::ALL || ((*it)->is_fermionic && ct == ComputeType::FERMIONIC) || (!(*it)->is_fermionic && ct == ComputeType::NONFERMIONIC))
```
Comment (PDoakORNL) [inline src/QMCWaveFunctions/TrialWaveFunction.cpp]: I think this is the absolute limit for how complex a conditional should need to be.

File: src/QMCHamiltonians/ECPotentialBuilder.cpp [hunk 2]
```diff
@@ -63,15 +63,23 @@ bool ECPotentialBuilder::put(xmlNodePtr cur)
 #ifdef ENABLE_OFFLOAD
   NLPP_algo = "batched"; // set "batched" as the default
 #endif
+  std::string useDLA("no");
```
Comment (PDoakORNL) [inline src/QMCHamiltonians/ECPotentialBuilder.cpp]: for new code we should use the new naming conventions.  use_DLA

Comment (prckent) [reply src/QMCHamiltonians/ECPotentialBuilder.cpp]: +1

**Comment by PDoakORNL** [comment]
Also I think the enum is fine.

File: src/QMCWaveFunctions/TrialWaveFunction.cpp (hunk 1 above)
Comment (ye-luo) [reply src/QMCWaveFunctions/TrialWaveFunction.cpp]: Do you want me to change something or this is just a comment?

File: src/QMCHamiltonians/ECPotentialBuilder.cpp (hunk 2 above)
Comment (ye-luo) [reply src/QMCHamiltonians/ECPotentialBuilder.cpp]: Will fix it soon.

File: src/QMCWaveFunctions/TrialWaveFunction.cpp (hunk 1 above)
Comment (PDoakORNL) [reply src/QMCWaveFunctions/TrialWaveFunction.cpp]: Just a comment. Although I think if we end up needing more logic here we should consider a different approach to dispatching this call.

Comment (PDoakORNL) [reply src/QMCWaveFunctions/TrialWaveFunction.cpp]: And the ComputeType tag sets us up to do that

**Comment by ye-luo** [comment]
@PDoakORNL variable name has been updated.

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**